src/main/java/org/pac4j/springframework/config/ResourceWrapper.java

```diff
@@ -0,0 +1,36 @@
+package org.pac4j.springframework.config;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.pac4j.core.io.Resource;
+
+/**
+ * A wrapper that adapts a spring {@link org.springframework.core.io.Resource} to a pac4j {@link Resource}.
+ * 
+ * @author Keith Garry Boyce
+ * @since 1.4.3
+ */
+public class ResourceWrapper implements Resource {

```

Comment:
OK. Now I have the big picture. I think we could need that in the `spring-webmvc-pac4j` or in `j2e-pac4j` libraries for example.

So you should move these two classes in your pull requests (in the `io`) package, rename them as `SpringResourceWrapper` and `Spring WritableResourceWrapper` with the Spring dependencies added as `optional` in pac4j.